crates/matrix-sdk-ui/src/room_list_service/room_list.rs

```diff
@@ -195,34 +195,37 @@ fn merge_stream_and_receiver(
 
         loop {
             select! {
-                biased; // Prefer manual updates for easier test code
+                // We want to give priority on updates from `raw_stream` as it will necessarily trigger a “refresh” of the rooms.
+                biased;
 
-                Ok(update) = roominfo_update_recv.recv() => {
-                    if !update.trigger_room_list_update {
-                        continue;
-                    }
+                diffs = raw_stream.next() => {
+                    tracing::error!("merge streams: receive diff");
```

Comment:
nit: debug printing here. If you want to keep it, lower the log level to trace.

Reply:
Removed.